### ISSUE TITLE
RST writer: images in links are ignored

### ISSUE THREAD

**Issue by mgeier** [jgm/pandoc#4810]
The following examples use this command line:

    pandoc -f markdown -t rst

... and this input text:

    [![alt-text](http://example.org/image.svg)](http://example.org/)

With `pandoc` version 2.1.3, I got this:

```
|alt-text|

.. |alt-text| image:: http://example.org/image.svg
   :target: http://example.org/
```

With `pandoc` versions 2.2.1 and 2.2.2, I'm getting this:

    `alt-text <http://example.org/>`__

Here is a rendered example where links in images work fine (`pandoc` version 2.1.3): https://nbsphinx.readthedocs.io/en/0.3.3/installation.html#nbsphinx-Packages

And here is the same page where the images are not shown (`pandoc` version 2.2.2): https://nbsphinx.readthedocs.io/en/0.3.4/installation.html#nbsphinx-Packages

**Comment by jgm** [comment]
@danse - is this an unintended consequence of your change
disallowing nested inline structures?

**Comment by danse** [comment]
hi, yes this is due to the code i wrote and i propose a fix in #4814, sorry for the inconvenience

**Comment by mgeier** [comment]
Thanks for the quick fix!
